src/coreComponents/codingUtilities/TableData.hpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file TableData.hpp
+ */
+
+#ifndef GEOS_COMMON_TableData_HPP
+#define GEOS_COMMON_TableData_HPP
+
+#include "common/DataTypes.hpp"
+
+namespace geos
+{
+
+class TableData
+{
+public:
+
+  /**
+   * @brief Add a row to the table.
+   * @param Args The values passed to addRow (can be any type).
+   * @param args Cell values to be added to the line.
+   */
+  template< typename ... Args >
+  void addRow( Args const & ... args );
```

Comment:
Maybe you can add a `clear()` method so the class users can reset their data each frames without internally reallocating.